Add support for subrepository scanning and regex tag matching in regmaid policies

libschmudie-so opened the pull request:
This PR introduces two main enhancements:

Subrepository Scanning:
Regmaid can now look for subrepositories within the configured repository when the hasSubrepos flag is enabled in a policy. This allows for more flexible and comprehensive repository management.

Regex Tag Matching:
Policies can now specify a regular expression for tag matching by setting the regex option to true. This enables advanced tag selection beyond simple wildcard patterns.

libschmudie-so committed:
allow config to contain an actual regex as an option

libschmudie-so committed:
allow regmaid to look for subrepositories withing the configured repository

moritzrinow commented:
@libschmudie-so thanks for showing interest in the project and taking time to contribute.

**Regarding Regex Tag Matching**:
This looks good to me. Super clear what's happening and useful. I'd merge this as is.

**Regarding Subrepository Scanning**:
This is a more complex one for me. Definitely thought about it too and I am definitely for it in general.

First of all, I would tend to avoid the additional config entry for `hasSubrepos` and instead activate the mechanism if there is at least one wildcard character present in the name. Example: `my-app/*`, which would match all repositories with that prefix. I think this should be fine because wildcard is not an actually allowed character for repository names, so it's clear that if you have  a wildcard in the repo name, you want to match multiple repositories.

Secondly, policies are meant to be processed in parallel. Before, this meant repository scanning happened in parallel, because it was 1 repo per policy. With your implementation as it is right now, repositories within a policy would be scanned sequentially.
I don't know if I like this. You could have one policy with `repository=*` and then there would be no parallelism whatsoever.

libschmudie-so commented:
I just rewrote the subrepo detection to use the same getRegex to match wildcards in the same style as the tags. I also moved the thread down to the actual processing.

moritzrinow commented:
I'll take it in, thanks a lot. Will make some minor adjustments improvements tho to the logging for it to make more sense now that there are N repos per policy. Also I think the repo listing should only be done if the repo name is a wildcard expression, as mentioned.
